AK/StringView.h

```diff
@@ -90,6 +90,9 @@ class StringView {
     Optional<size_t> find(const StringView&) const;
     Optional<size_t> find(char c) const;
 
+    void advance(size_t n);
+    StringView take_until_char_or_end(char c, bool consume_separator);
```

Comment:
This doesn't feel like a good fit for the `StringView` class, it's a super special mechanism for a single user. Can we find a better abstraction/place for this?

Reply:
Removed all the string manipulation stuff. It's using GenericLexer now.